File: core/qt_utils.py
from PyQt6.QtCore import QObject, pyqtSignal
import traceback


class MainThreadExecutor(QObject):
    """Schedule callables to run in the Qt main thread from any thread."""
    run = pyqtSignal(object, tuple, dict)

    def __init__(self, parent=None):
        super().__init__(parent)
        self.run.connect(self._call)

    def _call(self, func, args, kwargs):
        try:
            func(*args, **kwargs)
        except Exception:
            print("[MainThreadExecutor] Exception in scheduled function")
            traceback.print_exc()


_singleton = None


def get_main_thread_executor(parent=None):
    """Return a singleton MainThreadExecutor instance.

    The optional parent is ignored after the first creation.
    """
    global _singleton
    if _singleton is None:
        _singleton = MainThreadExecutor(parent)
    return _singleton

File: AudibleZenBot.AutoGen/core/qt_utils.cs
using System;
using System.Threading.Tasks;

namespace core.qt_utils {
    public static class Qt_utilsModule {
        // Original: def __init__(self, parent=None)
        public static void Init(object? parent = null) {
            // TODO: implement
        }

        // Original: def _call(self, func, args, kwargs)
        public static void Call(object? func, object? args, object? kwargs) {
            // TODO: implement
        }

        // Original: def get_main_thread_executor(parent=None)
        public static void GetMainThreadExecutor(object? parent = null) {
            // TODO: implement
        }

    }

    public class MainThreadExecutor {
        public object? run { get; set; }


        // Original: def __init__(self, parent=None)
        public MainThreadExecutor(object? parent = null) {
            // TODO: implement constructor
            this.run = null;
        }

        // Original: def _call(self, func, args, kwargs)
        public void Call(object? func, object? args, object? kwargs) {
            // TODO: implement
        }

        // Original: def get_main_thread_executor(parent=None)
        public void GetMainThreadExecutor(object? parent = null) {
            // TODO: implement
        }

    }

}

